core/communication_manager.py

```diff
@@ -0,0 +1,142 @@
+        self.ACTIONABLE_EVENT_TYPE_2_CALLBACK = {
+            ActionableEventTypes.SCHEDULE_JOB.value: self._schedule_job,
+            ActionableEventTypes.TERMINATE.value: self._terminate_session,
+            ActionableEventTypes.NOTHING.value: self._do_nothing,
+        }
+
+        # These are unmapped and will only be received when an optimizer is not
+        # active.
+        self.RAW_EVENT_TYPE_2_CALLBACK = {
+            RawEventTypes.NEW_SESSION.value: self._create_session,
+            RawEventTypes.NEW_WEIGHTS.value: self._do_nothing,
+            RawEventTypes.TERMINATE.value: self._do_nothing,
+            RawEventTypes.NOTHING.value: self._do_nothing,
+        }
+
+    def configure(self, scheduler):
+        """
+        Configures the Communication Manager so that it can submit jobs to the
+        Scheduler.
+        """
+        logging.info("Configuring the Communication Manager...")
+        self.scheduler = scheduler
+        logging.info("Communication Manager is configured!")
+
+    # Public methods
+    
+    def inform(self, event_type, payload):
+        """
+        Method called by other modules to inform the Communication Manager about
+        events that are going on in the service.
+
+        If there isn't an active optimizer, the commmgr will handle it
+
+        Else, payloads are relayed to the active optimizer which decides how 
+        and to whom the Communication Manager should communicate the event/message.
+
+        For example: A runner could inform the Communication Manager that the
+        node is done training a particular model, to which the Optimizer could
+        decide it's time to communicate the new weights to the network.
+        """
+        logging.info("Information has been received: type({})".format(event_type))
+        if self.optimizer:
+            event_type, payload = self.optimizer.ask(event_type, payload)
+            callback = callback_handler(event_type, self.ACTIONABLE_EVENT_TYPE_2_CALLBACK,
+            ActionableEventTypes.NOTHING.value)
+        else:
+            # We need to start a new session.
+            callback = callback_handler(event_type, self.RAW_EVENT_TYPE_2_CALLBACK,
+            RawEventTypes.NOTHING.value)
+        callback(payload)
+
+    # Callbacks
+
+    def _create_session(self, payload):
+        """
+        Creates a new session and optimizer based on the parameters sent by a
+        model developer. It then asks the optimizer what's the first thing to do
+        and the service starts working on that session from that point on.
+        """
+        # NOTE: We removed the 'optimizer_type' argument since for the MVP we're
+        # only considering the 'FederatedAveragingOptimizer' for now.
+        # TODO: We need to incorporate session id's when we're ready.
+        logging.info("New optimizer session is being set up...{}".format(payload))
+        # session_id = payload.get('key')
```

Comment:
I thought the 'key' was a blockchain-specific value, not the `session_id`?

@nzoghb @kiddyboots216 (Can we write a quick & dirty docs about this?)

Reply:
It'll get fixed when we merge in Blockchain Gateway and do integration testing, I'm sure.
Either way doesn't matter because it's in a comment.

Reply:
Ok. Will remove the commented code for now since it might be inaccurate.